conans/client/generators/cmake_find_package.py

```diff
@@ -43,52 +48,285 @@ class CMakeFindPackageGenerator(Generator):
+        set({{ pkg_name }}_VERSION "{{ pkg_version }}")
+
+        find_package_handle_standard_args({{ pkg_name }} REQUIRED_VARS
+                                          {{ pkg_name }}_VERSION VERSION_VAR {{ pkg_name }}_VERSION)
+        mark_as_advanced({{ pkg_name }}_FOUND {{ pkg_name }}_VERSION)
+
+
+        ########### VARIABLES #######################################################################
+        #############################################################################################
+
+        set({{ pkg_name }}_COMPONENTS {{ pkg_components }})
+        set({{ pkg_name }}_DEPENDENCIES {{ pkg_dependencies }})
+
+        {%- for comp_name, comp in components %}
+
+        ########### COMPONENT {{ comp_name }} VARIABLES #############################################
+
+        set({{ comp_name }}_INCLUDE_DIRS {{ comp.include_paths }})
+        set({{ comp_name }}_INCLUDE_DIR {{ comp.include_path }})
+        set({{ comp_name }}_INCLUDES {{ comp.include_paths }})
+        set({{ comp_name }}_LIB_DIRS {{ comp.lib_paths }})
+        set({{ comp_name }}_RES_DIRS {{ comp.res_paths }})
+        set({{ comp_name }}_DEFINITIONS {{ comp.defines }})
+        set({{ comp_name }}_COMPILE_DEFINITIONS {{ comp.compile_definitions }})
+        set({{ comp_name }}_COMPILE_OPTIONS_LIST "{{ comp.cxxflags_list }}" "{{ comp.cflags_list }}")
+        set({{ comp_name }}_LIBS {{ comp.libs }})
+        set({{ comp_name }}_SYSTEM_LIBS {{ comp.system_libs }})
+        set({{ comp_name }}_FRAMEWORK_DIRS {{ comp.framework_paths }})
+        set({{ comp_name }}_FRAMEWORKS {{ comp.frameworks }})
+        set({{ comp_name }}_BUILD_MODULES_PATHS {{ comp.build_modules_paths }})
+        set({{ comp_name }}_DEPENDENCIES {{ comp.public_deps }})
+
+        {%- endfor %}
+
+
+        ########## FIND PACKAGE DEPENDENCY ##########################################################
+        #############################################################################################
+
+        include(CMakeFindDependencyMacro)
+
+        {%- for public_dep in pkg_public_deps %}
+
+        if(NOT {{ public_dep }}_FOUND)
+            find_dependency({{ public_dep }} REQUIRED)
+        else()
+            message(STATUS "Dependency {{ public_dep }} already found")
+        endif()
+
+        {%- endfor %}
+
+
+        ########## FIND LIBRARIES & FRAMEWORKS / DYNAMIC VARS #######################################
+        #############################################################################################
+
+        {%- for comp_name, comp in components %}
+
+        ########## COMPONENT {{ comp_name }} FIND LIBRARIES & FRAMEWORKS / DYNAMIC VARS #############
+
+        set({{ comp_name }}_LIB_TARGETS "") # Will be filled later, if CMake 3
+        conan_component_library_targets({{ comp_name }}_LIB_TARGETS "${% raw %}{{% endraw %}{{ comp_name }}_LIB_DIRS{% raw %}}{% endraw %}" "${% raw %}{{% endraw %}{{ comp_name }}_LIBS{% raw %}}{% endraw %}")
```

Comment:
Did not found a way to make things like this ``${{{ comp_name }}_LIBS}` work with jinja2 and had to use ``{% raw %}{{% endraw %}``, is there any other way?

Reply:
maybe you can use: `{{ '{'+comp_name+'}' }}`.

Reply:
Don't get it, Like `{{'${'+comp_name+'_LIBS}'}}` ?

Reply:
Yes, inside the braces `{{ <here> }}` you can build the string you need.

Reply:
ok, I have to try this

Reply:
It works! Thanks 🥇